### ISSUE TITLE
Bringing templates to GitHawk

### ISSUE THREAD

**Issue by Huddie** [GitHawkApp/GitHawk#2393]
Working on this, if anyone has ideas or suggestions please lmk!

<img src=https://user-images.githubusercontent.com/10662471/47959045-1dae4b00-dfb0-11e8-9072-6a8a15069696.gif width=200>

**Comment by rnystrom** [comment]
Yo this is so cool!! Would this work for the “new issue” feature in repos too?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Huddie** [comment]
@rnystrom I’m almost positive! I checked github and all issues are under .github/ISSUE_TEMPLATE 

Right now I fetch for specific files but my end plan is fetch all files under that directory, parse the text for the name / description 

Should work for all repos! 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Huddie** [comment]
@rnystrom Confirmed that it will work across the board. Done for the night 🌙

<img src=https://user-images.githubusercontent.com/10662471/47960194-3a0cb080-dfcd-11e8-9a09-8354aef96c8d.gif width=200>

**Comment by Huddie** [comment]
@Sherlouk the old one being in root/docs/.github rather than strictly .github? I’m not so familiar with github templates, searching online It seemed that you could have as many templates as you want but must be in .github/ISSUE_TEMPLATE directory. Is this not true, can you have templates in other places? Quick note: I’m speaking specifically about issue templates not PR templates 

I think for now it’s fine to support this way as if no templates are found it simply opens up the new issue view like it used to. We can expand to support other templates in the future, if others exist.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Sherlouk** [comment]
@Huddie Yea so I think the difference is with the new system you can have multiple named templates but previously you could have a single template that was called `ISSUE_TEMPLATE.md` (and could be in a multitude of places) and would always use that one.

Essentially I think the chain of logic is:

1. Check for templates in `.github/ISSUE_TEMPLATE/` directory
2. Check for single template called `ISSUE_TEMPLATE.md` in either root or `.github` (name not case sensitive)
3. Just leave it empty

I think for the sake of this, just doing that first one is definitely great progress!

**Comment by Huddie** [comment]
Thanks @Sherlouk I will PR number 1 and then maybe expand after that. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ijm8710** [comment]
Really cool!!!

**Comment by ijm8710** [comment]
Me: Curious, so the way it works is to provide a framework on what you’re reporting? Is it the user’s job to replace the template with their report? Is there any way to set it up so it prompts you, this is what you enter here without the need to delete? This is super cool either way but was just curious so that the template is still a reference but also streamlines the input

Huddies reply to me: if you could move this comment to #2393 this thread is more for the release of the feature, that thread is about the feature itself. Quick answer is, templates are a Github thing so we are mimicking the feature as much as possible. No fill ins etc. Just a template of what to write. Unfortunately that means deleting text.

**Comment by Huddie** [comment]
It could be one day we expand the feature natively

**Comment by ijm8710** [comment]
Yeah understood this is more baseline and it seems awesome. Was just curious if you were on same page as me that this would even have more utility with my suggestion and it sounds like over time it could add a little value

**Comment by superandrew** [comment]
this is so cool! is it still on the works? or is it available already in some distribution/beta testing of the app?
